### PR TITLE
Trait: FieldPosition

### PR DESCRIPTION
**Pull request by ax3l**
Refactors the getter of field positions into a trait. Read commit-wise (commit 1: refactor, commit 2: update usage in code).

Adds the previously just commented positions of FieldTmp (currently used in ParticleToGrid always at zero corner.)

Used in the upcoming openPMD branch.

### CODE REVIEW

File: src/picongpu/include/fields/numericalCellTypes/EMFCenteredCell.hpp [hunk 1]
```diff
@@ -23,122 +23,139 @@
...
 namespace picongpu
 {
-using namespace PMacc;
 namespace emfCenteredCell
 {
+namespace traits
+{
+    /** \tparam floatD_X position of the component in the cell
+     *  \tparam DIM3     Fields (E/B/J) have 3 components, even in 1 or 2D ! */
+    //typedef ::PMacc::math::Vector<floatD_X,DIM3> VectorVector;
```
Comment (psychocoderHPC) [inline src/picongpu/include/fields/numericalCellTypes/EMFCenteredCell.hpp]: please remove

File: src/picongpu/include/fields/numericalCellTypes/EMFCenteredCell.hpp [hunk 2]
```diff
@@ -23,122 +23,139 @@
...
 
 namespace picongpu
 {
-using namespace PMacc;
 namespace emfCenteredCell
 {
+namespace traits
+{
+    /** \tparam floatD_X position of the component in the cell
+     *  \tparam DIM3     Fields (E/B/J) have 3 components, even in 1 or 2D ! */
```
Comment (psychocoderHPC) [inline src/picongpu/include/fields/numericalCellTypes/EMFCenteredCell.hpp]: maybe the closing tag on a new line to support better readability without code highlighting

**Comment by psychocoderHPC** [comment]
well done

**Comment by psychocoderHPC** [comment]
Could you please run one LWFA or KHI test?

**Comment by ax3l** [comment]
I did run those already in my openPMD branch with KHI

**Comment by psychocoderHPC** [comment]
offline discussed with @ax3l : Tests looks correct

**Pull request merged**